Assets/Scripts/Models/Health/Health.cs

```diff
@@ -0,0 +1,227 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software,
+// and you are welcome to redistribute it under certain conditions; See
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using MoonSharp.Interpreter;
+
+[MoonSharpUserData]
+[Serializable]
+public class Health
+{
+    //// Health States Order
+    //// Revive, Heal, Overheal, Hit, Destroyed
+
+    private float currentHealth;
+
+    public Health(float maxHealth)
+    {
+        MaxHealth = maxHealth;
+        currentHealth = maxHealth;
+    }
+
+    public Health(float maxHealth, bool isInvincible, bool isHealable, bool isRevivable, bool canOveheal)
+    {
+        MaxHealth = maxHealth;
+        currentHealth = maxHealth;
+        IsInvincible = isInvincible;
+        IsHealable = isHealable;
+        IsRevivable = isRevivable;
+        CanOverheal = canOveheal;
+    }
+
+    private Health(Health other)
+    {
+        currentHealth = other.currentHealth;
+        MaxHealth = other.MaxHealth;
+        IsInvincible = other.IsInvincible;
+        IsHealable = other.IsHealable;
+        IsRevivable = other.IsRevivable;
+        CanOverheal = other.CanOverheal;
+    }
+
+    public event Action Revive;
+
+    public event Action Heal;
+
+    // TODO: Will be used to notify user if healing using something like a medkit will heal for X amount with some wasted (because we hit maxHP)
+    public event Action Overheal;
+
+    public event Action Hit;
+
+    public event Action Destroyed;
+
+    public float CurrentHealth
+    {
+        get
+        {
+            return currentHealth;
+        }
+
+        set
+        {
+            // New health greater than current health.
+            if (value > currentHealth)
+            {
+                EvaluateHealthIncrease(value);
+            }
+
+            // New health less than current health.
```

Comment:
yeah but its inside a if..elseif...else logic. stylecop wants it on top of the first if, so line 67.
just prefix it with // Else:

and yeah i agree that the rule is unnessicary and unclear inside of if else statements.